Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithInProcessRenderingBackingStore.mm

```diff
@@ -254,15 +261,17 @@ void flushAndCollectHandles(HashMap<RemoteImageBufferSetIdentifier, std::unique_
     }
 }
 
-RefPtr<WebCore::ImageBuffer> RemoteLayerWithInProcessRenderingBackingStore::allocateBuffer() const
+RefPtr<WebCore::ImageBuffer> RemoteLayerWithInProcessRenderingBackingStore::allocateBuffer()
 {
     auto purpose = m_layer->containsBitmapOnly() ? WebCore::RenderingPurpose::BitmapOnlyLayerBacking : WebCore::RenderingPurpose::LayerBacking;
     ImageBufferCreationContext creationContext;
     creationContext.surfacePool = &WebCore::IOSurfacePool::sharedPool();
 
 #if ENABLE(RE_DYNAMIC_CONTENT_SCALING)
-    if (m_parameters.includeDisplayList == IncludeDisplayList::Yes)
+    if (m_parameters.includeDisplayList == IncludeDisplayList::Yes) {
+        creationContext.dynamicContentScalingResourceCache = ensureDynamicContentScalingResourceCache();
```

Comment:
Maybe calling `ensureDynamicContentScalingResourceCache()` can be moved to `RemoteLayerWithInProcessRenderingBackingStore::ensureBackingStore()`. So this one can just assign 
```
creationContext.dynamicContentScalingResourceCache = m_dynamicContentScalingResourceCache;
```
And the function can stay const.